internal/lsp/server.go

```diff
@@ -989,7 +989,9 @@ func (l *LanguageServer) StartTemplateWorker(ctx context.Context) {
 			// determine the new contents for the file, if permitted
 			newContents, err := l.templateContentsForFile(job.URI)
 			if err != nil {
-				l.logError(fmt.Errorf("failed to template new file: %w", err))
+				if !errors.Is(err, &templatingInRootError{}) {
+					l.logError(fmt.Errorf("failed to template new file: %w", err))
```

Comment:
Is it really an error unless it's something the user asked for and didn't get? If I saw this in the logs, I would probably think I did something wrong and that I should look into how to fix it.

Reply:
I've opted to not use the error for control flow here now: 0936ca7b7827ed2fc91a93a3b00bbccf7556e3be


While it's still an error, the callers of the function are now expected to test if the file is in the root and avoid calling the function accordingly.